### PR TITLE
Add option to specify MultiRenderTarget's depth texture format.

### PR DESCRIPTION
**Pull request by kircher1**
For WebGL 2, this change allows for choosing the depth texture format (16, 24, 24_S8, or 32F). Default is the same as before: 16 bits.

As an optimization, glRenderBuffers do not need to be created if a depth(/stencil) texture is being generated since it will be put on the framebuffer's depth/stencil attachment instead.

### CODE REVIEW

File: src/Engines/Extensions/engine.multiRender.ts [hunk 1]
```diff
@@ -185,6 +186,12 @@ ThinEngine.prototype.createMultipleRenderTarget = function (size: RenderTargetTe
         if (options.samplingModes) {
             samplingModes = options.samplingModes;
         }
+        if (this.webGLVersion > 1 &&
+            (options.depthTextureFormat == Constants.TEXTUREFORMAT_DEPTH24_STENCIL8 ||
```
Comment (deltakosh) [inline src/Engines/Extensions/engine.multiRender.ts:190]: ===

**Review by deltakosh — APPROVED** [review]
looks good to me

File: src/Engines/Extensions/engine.multiRender.ts [hunk 2]
```diff
@@ -198,12 +205,13 @@ ThinEngine.prototype.createMultipleRenderTarget = function (size: RenderTargetTe
...
-    var depthStencilBuffer = this._setupFramebufferDepthAttachments(generateStencilBuffer, generateDepthBuffer, width, height);
+    var useStencilTexture = this.webGLVersion > 1 && generateDepthTexture && options.depthTextureFormat == Constants.TEXTUREFORMAT_DEPTH24_STENCIL8;
+    var depthStencilBuffer = this._setupFramebufferDepthAttachments(!useStencilTexture && generateStencilBuffer, !generateDepthTexture && generateDepthBuffer, width, height);
 
     rtWrapper._framebuffer = framebuffer;
     rtWrapper._depthStencilBuffer = depthStencilBuffer;
-    rtWrapper._generateDepthBuffer = generateDepthBuffer;
-    rtWrapper._generateStencilBuffer = generateStencilBuffer;
+    rtWrapper._generateDepthBuffer = !generateDepthTexture && generateDepthBuffer;
+    rtWrapper._generateStencilBuffer = !generateDepthTexture && generateStencilBuffer;
```
Comment (kircher1) [inline src/Engines/Extensions/engine.multiRender.ts:214]: meant this to be: !useStencilTexture

File: src/Engines/constants.ts [hunk 3]
```diff
@@ -164,6 +164,8 @@ export class Constants {
   public static readonly TEXTUREFORMAT_DEPTH32_FLOAT = 14;
   /** Depth 16 bits */
   public static readonly TEXTUREFORMAT_DEPTH16 = 15;
+  /** Depth 24 bits */
+  public static readonly TEXTUREFORMAT_DEPTH24 = 16;
```
Comment (Popov72) [inline src/Engines/constants.ts:168]: You need to also update the file components/tabs/propertyGrids/materials/texturePropertyGridComponent.tsx of the inspector to add support for this new type of format (see around line 63).

File: src/Engines/Extensions/engine.multiRender.ts [hunk 4]
```diff
@@ -198,12 +205,13 @@ ThinEngine.prototype.createMultipleRenderTarget = function (size: RenderTargetTe
...
 
-    var depthStencilBuffer = this._setupFramebufferDepthAttachments(generateStencilBuffer, generateDepthBuffer, width, height);
+    var useStencilTexture = this.webGLVersion > 1 && generateDepthTexture && options.depthTextureFormat == Constants.TEXTUREFORMAT_DEPTH24_STENCIL8;
+    var depthStencilBuffer = this._setupFramebufferDepthAttachments(!useStencilTexture && generateStencilBuffer, !generateDepthTexture && generateDepthBuffer, width, height);
 
     rtWrapper._framebuffer = framebuffer;
     rtWrapper._depthStencilBuffer = depthStencilBuffer;
-    rtWrapper._generateDepthBuffer = generateDepthBuffer;
-    rtWrapper._generateStencilBuffer = generateStencilBuffer;
+    rtWrapper._generateDepthBuffer = !generateDepthTexture && generateDepthBuffer;
```
Comment (kircher1) [inline src/Engines/Extensions/engine.multiRender.ts:213]: If I'm interpreting the spec correctly, there's no point in generating a stencil buffer here either:

    FRAMEBUFFER_UNSUPPORTED if depth and stencil attachments, if present, are not the same image. 

3.7.4 https://www.khronos.org/registry/webgl/specs/2.0/

Comment (kircher1) [reply src/Engines/Extensions/engine.multiRender.ts:213]: i.e., both _generateDepthBuffer and _generateStencilBuffer should be gated on the same condition (!_generateDepthTexture)

**Review by Popov72 — APPROVED** [review]
LGTM.

However, you will need to remove the dist/preview release/babylon.module.d.ts file from the PR.

**Comment by sebavan** [comment]
@kircher1 I ll merge as soon as you ll revert the changes to the babylon/module.d.ts file :-) Thanks a lot for the change

File: src/Engines/Extensions/engine.multiRender.ts (hunk 4 above)
Comment (kircher1) [reply src/Engines/Extensions/engine.multiRender.ts:213]: @sebavan if you have any thoughts on this let me know. I'm actually going to leave it how it is in the PR. I realized if I disable the stencil render buffer when depth texture is enabled, that's a breaking change as previous code allowed for this too. And FWIW, I haven't been able to repro any GL errors by binding a stencil render buffer and a depth(/stencil) texture at the same time, so it's unclear to me if that is actually against the spec or not.

**Pull request merged**